kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/RollingUpdateProcessServiceIntegrationTest.java

```diff
@@ -254,4 +277,183 @@ public void testStartProcessInDifferentDeploymentWithContainerId() throws Except
         taskClient.completeAutoProgress(CONTAINER_ALIAS, task.getId(), USER_YODA, null);
 
     }
+
+    @Test
+    public void testDifferentGAVsWithAlias() throws Exception {
+        Long oldPid = processClient.startProcess(CONTAINER_ALIAS, PROCESS_ID_EVALUATION);
+        assertNotNull(oldPid);
+        assertTrue(oldPid > 0);
+
+        ProcessInstance processInstance = processClient.getProcessInstance(CONTAINER_ALIAS, oldPid);
+        assertNotNull(processInstance);
+        assertEquals(CONTAINER_ID, processInstance.getContainerId());
+
+        // Create a container with completely different GAV
+        ReleaseId differentReleaseId101 = new ReleaseId("org.kie.server.different", "different-gav-definition-project",
+                "1.0.1.Final");
+        KieContainerResource containerResource = new KieContainerResource(CONTAINER_ID_101, differentReleaseId101);
+        containerResource.setContainerAlias(CONTAINER_ALIAS);
+        client.createContainer(CONTAINER_ID_101, containerResource);
+
+        Long pid = processClient.startProcess(CONTAINER_ALIAS, PROCESS_ID_EVALUATION_2);
+        assertNotNull(pid);
+        assertTrue(pid > 0);
+
+        processInstance = processClient.getProcessInstance(CONTAINER_ALIAS, pid);
+        assertNotNull(processInstance);
+        assertEquals(CONTAINER_ID_101, processInstance.getContainerId());
+
+    }
+
+    @Test
+    public void testContainerIdAsAlias() throws Exception {
+        Long oldPid = processClient.startProcess(CONTAINER_ALIAS, PROCESS_ID_EVALUATION);
+        assertNotNull(oldPid);
+        assertTrue(oldPid > 0);
+
+        ProcessInstance processInstance = processClient.getProcessInstance(CONTAINER_ALIAS, oldPid);
+        assertNotNull(processInstance);
+        assertEquals(CONTAINER_ID, processInstance.getContainerId());
+
+        // Create the second container with alias equal to the first containerId
+        KieContainerResource containerResource = new KieContainerResource(CONTAINER_ID_101, releaseId101);
+        containerResource.setContainerAlias(CONTAINER_ID);
+        client.createContainer(CONTAINER_ID_101, containerResource);
+
+        try {
+            processClient.startProcess(CONTAINER_ID, PROCESS_ID_EVALUATION_2);
```

Comment:
Please use RestJmsSharedBaseIntegrationTest.assertClientException for negative tests to check response HTTP status and error messages.